### PR TITLE
[Merged by Bors] - feat: the additive circle is path connected

### PR DESCRIPTION
**Pull request by ocfnash**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Instances/AddCircle.lean [hunk 1]
```diff
@@ -505,6 +506,11 @@ end LinearOrderedField
 
 variable (p : ℝ)
 
+instance pathConnectedSpace : PathConnectedSpace $ AddCircle p := by
```
Comment (urkud) [inline Mathlib/Topology/Instances/AddCircle.lean:509]: I think that we should have a theorem `Function.Surjective.pathConnectedSpace` and an instance `PathConnectedSpace (Quotient _)`.

**Comment by urkud** [comment]
We have (at least) 2 circles. Please specify that this PR is about `AddCircle` in the commit message.

File: Mathlib/Topology/Instances/AddCircle.lean (hunk 1 above)
Comment (ocfnash) [reply Mathlib/Topology/Instances/AddCircle.lean:509]: Good idea, done.

File: Mathlib/Topology/Connected.lean [hunk 2]
```diff
@@ -763,6 +770,15 @@ theorem isConnected_range [TopologicalSpace β] [ConnectedSpace α] {f : α →
   ⟨range_nonempty f, isPreconnected_range h⟩
 #align is_connected_range isConnected_range
 
+theorem Function.Surjective.connectedSpace [ConnectedSpace α] [TopologicalSpace β]
+  {f : α → β} (hf : Surjective f) (hf' : Continuous f) : ConnectedSpace β := by
+  rw [connectedSpace_iff_univ, ← range_iff_surjective.mpr hf]
```
Comment (urkud) [inline Mathlib/Topology/Connected.lean:775]: Micro golfing:
```suggestion
  rw [connectedSpace_iff_univ, ← hf.range_eq]
```

Comment (urkud) [reply Mathlib/Topology/Connected.lean:775]: Another option: add `DenseRange.connectedSpace` by reusing `DenseRange.preconnectedSpace`, then use `Function.Surjective.denseRange`.

File: Mathlib/Topology/PathConnected.lean [hunk 3]
```diff
@@ -1142,6 +1142,29 @@ theorem pathConnectedSpace_iff_univ : PathConnectedSpace X ↔ IsPathConnected (
...
+  pathConnectedSpace_iff_univ.mp inferInstance
+
+theorem isPathConnected_range [PathConnectedSpace X] {f : X → Y} (hf : Continuous f) :
+    IsPathConnected (range f) := by
+  rw [← image_univ]
+  exact isPathConnected_univ.image hf
+
+theorem Function.Surjective.pathConnectedSpace [PathConnectedSpace X]
+  {f : X → Y} (hf : Surjective f) (hf' : Continuous f) : PathConnectedSpace Y := by
+  rw [pathConnectedSpace_iff_univ, ← range_iff_surjective.mpr hf]
```
Comment (urkud) [inline Mathlib/Topology/PathConnected.lean:1155]: Same here:
```suggestion
  rw [pathConnectedSpace_iff_univ, ← hf.range_eq]
```

**Comment by urkud** [comment]
Otherwise LGTM. Thanks!
bors d+

**Comment by ocfnash** [comment]
bors merge

**Pull request closed without merging**